.github/workflows/build.yml

```diff
@@ -98,6 +98,7 @@ jobs:
 
     steps:
       - name: Free up space
+        if: runner.name != 'arc-runner-set'
```

Comment:
can we factor out this condition somehow? e.g., have one step that defines a true/false condition, and use that in the following steps?
Right now "if arc-runner-set" is correct, but the same skipping of prerequisites installation is needed for #1055 but slightly different.